File: ios/vwa_mms_app/I18nSystemInfo.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(I18nSystemInfo, NSObject)

RCT_EXTERN_METHOD(getSystemLanguage:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/vwa_mms_app/I18nSystemInfo.swift
import Foundation
import React

@objc(I18nSystemInfo)
class I18nSystemInfo: NSObject {

  @objc
  func getSystemLanguage(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    let language = Locale.preferredLanguages.first ?? "en"
    resolve(language)
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
